docs/Course Wiki/ECE Course Offerings/ECE374B.md

```diff
@@ -0,0 +1,66 @@
+# ECE374B
+## Introduction to Algorithms and Models of Computation
+
+ECE374B is one of two **independent** sections of CS/ECE374, Introduction to Algorithms and Models of Computation. It is unoficially the 'ECE' version of 374. It is considered by some to be easier than the A version.
+
+## Content Covered
+
+#### Theory of Computation 
+- Regular and context-free languages.
+- Fooling sets.
+- Determinisitc and non-deterministic finite state automata.
+- Push down automata
+- Turing Machines
+- Computational Complexity
+#### Algorithms
+- Divide & Conquer
+- Graph Algorithms
+- Shortest Paths, MSTs
+- Dynamic Programming
+- Directed Acylic Graphs
+- Algorithmic Complexity Analysis
+#### Secret Third Thing
+- Reductions
+- Undecidability
+- NP, NP-Hardness, P vs NP
+
+
+## Prerequisites
+- [CS 173](../CS%20Course%20Offerings/CS173.md)
+- [CS 225](../CS%20Course%20Offerings/CS225.md)
+
+There are two official prerequisites to ECE 374, CS 173 - Discrete Structures, and CS 225 - Data Structures and Algorithms. 374B expands on the theoretical computer science from CS173 and covers algorithms more in depth than CS225, without any of the programming. For part one of the course, you should remember induction, set notation, and FSMs from CS 173. Induction and recursion are fundamental to the course, so it is important to know them well. In the second portion of the course, there is a significant overlap with the graph algorithms taught in CS225.
+
+## 374B vs 374A
+There are two versions of 374 offered at UIUC, section A and section B. They have different lectures, different discussions, different homeworks and exams, different grading. Most people consider the B section to be less difficult than the A section. One can check the [grade disparity](https://waf.cs.illinois.edu/discovery/grade_disparity_between_sections_at_uiuc/) to see the actual GPA differences. The B section lacks the emphasis on formal mathematical proof writing that the A section has. The A section and B section have different amounts of midterms and different grading policies. Read the [CS 374A page](../CS%20Course%20Offerings/ECE374A.md) as well to learn more about the differences.
```

Comment:
Add this do the X vs Y section plz I wanna try to keep all the classes uniform in terms of headers